aptos-labs/aptos-core
# [indexer][protobuf] Move decoding PB transaction to data service level

### Description
Currently before we push transaction data into protobuf, we encode the transaction data into a base64 string. However this requires gRPC clients to decode the protobuf data twice (once from base64 and once again to deserialize the protobuf data). To avoid this double decoding on the client side, we can store the transaction data in the protobuf as bytes instead of a base64 string and 

Note: before inserting the transaction data into cache & filestore, we do need to base64 encode it in order to be compatible with JSON. When we read the transaction data from cache or filestore we do need to base64 decode in the gRPC data service. 

### Test Plan
1. Create local redis instance
2. Start local fullnode
3. Connect local cache worker to local redis
4. Start local data service
5. Verify parsed transactions are correct 
6. `buf breaking` shows that there's breaking change, which is fine because there's only 1 partner using the gRPC endpoint right now and we'll communicate with them about this change. In the future, we should add `buf breaking` into CI/CD pipeline.  
```
reneetso@Renees-MBP aptos-protos % buf breaking proto --against "../../.git#branch=origin/main,subdir=crates/aptos-protos/proto"           
proto/aptos/datastream/v1/datastream.proto:23:3:Field "1" on message "TransactionOutput" changed type from "string" to "bytes".
```
